hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4282,3 +4301,79 @@ func (r HostedClusterReconciler) getPayloadImage(ctx context.Context, hc *hyperv
 	}
 	return image, nil
 }
+
+func (r *HostedClusterReconciler) reconcileServiceAccountSigningKey(ctx context.Context, hc *hyperv1.HostedCluster, targetNamespace string, createOrUpdate upsert.CreateOrUpdateFN) error {
+	privateBytes, publicBytes, err := r.serviceAccountSigningKeyBytes(ctx, hc)
+	if err != nil {
+		return err
+	}
+	cpSigningKeySecret := controlplaneoperator.ServiceAccountSigningKeySecret(targetNamespace)
+	_, err = createOrUpdate(ctx, r.Client, cpSigningKeySecret, func() error {
+		// Only set the signing key when the key does not already exist
+		if cpSigningKeySecret.ResourceVersion != "" && cpSigningKeySecret.Data != nil {
```

Comment:
why the `ResourceVersion` check?

Reply:
Because if there isn't a ResourceVersion, then I'm creating the secret and I don't need to check for an existing key. However, I see that it's redundant (cpSigningKeySecret.Data != nil does the same thing). I can remove.

Reply:
fixed